FSI/rigid_flap/OpenFOAM-RigidBody/Solid/rigid_body_solver.cpp

```diff
@@ -0,0 +1,246 @@
+#include <iostream>
+#include <math.h>
+#include <precice/SolverInterface.hpp>
+
+using Vector = std::vector<double>;
+
+struct DataContainer {
+  void save_old_state(const Vector &vertices,
+                      const double &theta,
+                      const double &theta_dot)
+  {
+    old_vertices  = vertices;
+    old_theta     = theta;
+    old_theta_dot = theta_dot;
+  }
+
+  void reload_old_state(Vector &vertices,
+                        double &theta,
+                        double &theta_dot) const
+  {
+    vertices  = old_vertices;
+    theta     = old_theta;
+    theta_dot = old_theta_dot;
+  }
+
+  Vector old_vertices;
+  double old_theta;
+  double old_theta_dot;
+};
+
+class Solver {
+public:
+  Solver(const double moment_of_inertia)
+      : moment_of_inertia(moment_of_inertia)
+  {
+  }
+
+  void
+  solve(const Vector &forces,
+        const Vector &initial_vertices,
+        Vector &      vertices,
+        double &      theta,
+        double &      theta_dot,
+        const double  delta_t) const
+  {
+    // Compute total moment M = x^{n} x f^{n+1}
+    double moment = 0;
+    for (uint i = 0; i < forces.size() / 2; ++i)
```

Comment:
Just use `auto` here (where does actually `uint` come from?)

```suggestion
    for (auto i = 0; i < forces.size() / 2; ++i)
```

Reply:
`unsigned int` ?

Reply:
It looks like `uint` is compiler-specific: https://stackoverflow.com/questions/3552094/c-uint-unsigned-int-int

Reply:
Oh, I was not aware, good to know